runtime/builtin_types.go

```diff
@@ -193,6 +193,81 @@ func builtinAbs(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 	return Abs(f, args[0])
 }
 
+func zipLongestImpl(f *Frame, args Args) ([]*Object, *BaseException) {
+	argc := len(args)
+	result := make([]*Object, 0, 2)
+	iters := make([]*Object, argc)
+	for i, arg := range args {
+		iter, raised := Iter(f, arg)
+		if raised != nil {
+			return nil, raised
+		}
+		iters[i] = iter
+	}
```

Comment:
Seems like you could factor this iterator collecting logic out of map() and zip() into a function that returns []*Object. But that could wait until a later PR if you prefer.